File: ibkr_show_backend/app/schemas/earnings_settings.py
from pydantic import BaseModel


class EarningsPushSettings(BaseModel):
    enabled: bool = False
    push_time: str = "09:00"
    smtp_server: str = ""
    smtp_port: int = 587
    smtp_username: str = ""
    smtp_password: str = ""
    sender_email: str = ""
    target_email: str = ""


class TestSendRequest(BaseModel):
    smtp_server: str
    smtp_port: int = 587
    smtp_username: str
    smtp_password: str
    sender_email: str
    target_email: str


class TestSendResponse(BaseModel):
    success: bool
    message: str

File: ibkr_show_frontend/src/api/earningsSettings.ts
import { request } from './http'

export interface EarningsPushSettings {
  enabled: boolean
  push_time: string
  smtp_server: string
  smtp_port: number
  smtp_username: string
  smtp_password: string
  sender_email: string
  target_email: string
}

export interface TestSendResult {
  success: boolean
  message: string
}

export function fetchPushSettings(): Promise<EarningsPushSettings> {
  return request<EarningsPushSettings>('/api/earnings-settings/push')
}

export function updatePushSettings(settings: EarningsPushSettings): Promise<EarningsPushSettings> {
  return request<EarningsPushSettings>('/api/earnings-settings/push', {
    method: 'PUT',
    body: JSON.stringify(settings),
  })
}

export function sendTestEmail(settings: EarningsPushSettings): Promise<TestSendResult> {
  return request<TestSendResult>('/api/earnings-settings/push/test-send', {
    method: 'POST',
    body: JSON.stringify(settings),
  })
}

export function triggerDailyPush(): Promise<TestSendResult> {
  return request<TestSendResult>('/api/earnings-settings/push/trigger', {
    method: 'POST',
  })
}
